classifiers/reference_quality/special_character_classifier/__init__.py

```diff
@@ -0,0 +1,37 @@
+import unicodedata
+from typing import Optional, List, Tuple
+from pydantic import BaseModel
+
+INPUT_EXAMPLE = {
+    "text": "Super funny haha 😀.",
+    "allowedRanges": None
+}
+
+ALLOWED_RANGES = set(range(0x0020, 0x007F)).union( # Basic Latin
+    set(range(0x00A0, 0x00FF)), # Latin-1 Supplement
+    set(range(0x0100, 0x017F)),  # Latin Extended-A
+    set(range(0x0180, 0x024F)),  # Latin Extended-B
+    set(range(0x2000, 0x206F)),  # General Punctuation
+    set(range(0x20A0, 0x20CF)),  # Currency Symbols
+    set([ord("\t"), ord("\n"), ord("\r")])# common stop chars
+    )  
+
+class SpecialCharacterClassifierModel(BaseModel):
+    text: str
+    allowed_ranges: Optional[List[Tuple[int,int]]] = None
+
+    class Config:
+        schema_extra = {"example": INPUT_EXAMPLE}
+
+
+def special_character_classifier(req: SpecialCharacterClassifierModel):
+    """Checks if a string contains special characters"""
+    text = req.text
+    allowed_ranges = req.allowed_ranges
+    if allowed_ranges is None:
+        allowed_ranges = ALLOWED_RANGES
```

Comment:
i dont think this will work as expected. allowed ranges is a list of tuple (from previous version). The default is a single set with all values. Further down the validation is a simple in with would never fit imo. 

Maybe we should create a small help function that handels the check:

``` python

from typing import Union, Tuple, List, Set

def is_in_range(value:int, ranges:Union[Tuple[int,int],Set[int],List[Union[Tuple[int,int],Set[int]]]]):
    if isinstance(ranges,list):
        for t in ranges:
            if is_in_range(value,t):
                return True
        return False
    elif isinstance(ranges,tuple):
        return ranges[0] <= value <= ranges[1]
    elif isinstance(ranges,set):
        return value in ranges
    return False
        
# tester code
RANGE_SET = {1,2,3,6,7,8,9}
RANGE_TUPLE = (1,4)
RANGE_LIST_TUPLE = [(1,3),(6,9)]
RANGE_LIST_SET = [{1,2,3},{6,7,8,9}]

for value in  range(10):
    print(value, "is in range S :", is_in_range(value,RANGE_SET))
    print(value, "is in range T :", is_in_range(value,RANGE_TUPLE))
    print(value, "is in range LT:", is_in_range(value,RANGE_LIST_TUPLE))
    print(value, "is in range LS:", is_in_range(value,RANGE_LIST_SET))
```

Reply:
alternativly we could only allow sets as parameter